### PR TITLE
Add Unicode normalization and IDNA encoding to qute-pass userscript

### PR DESCRIPTION
**Pull request by DrTobe**
I have implemented support for Internationalized Domain Names (IDNA). Currently, this only kicks in when the normal 'pass' mode (not 'gopass') is used because I am no gopass user so I could not test this right now.

@cryzed You are listed as script author, would you mind having a look at this?

### CODE REVIEW

**Comment by DrTobe** [comment]
@The-Compiler Would you mind having a look at this?

**Comment by The-Compiler** [comment]
I've never used either qute-pass or pass, so I can't say much. Maybe @bascht or @lufte have an opinion on this?

**Comment by cryzed** [comment]
I don't use qutebrowser or qute-pass anymore, sorry.

**Comment by lufte** [comment]
I can take a look later.

File: misc/userscripts/qute-pass [hunk 1]
```diff
@@ -140,9 +159,13 @@ def find_pass_candidates(domain, unfiltered=False):
             split_path = pass_path.split(os.path.sep)
             for secret in secrets:
                 secret_base = os.path.splitext(secret)[0]
-                if not unfiltered and domain not in (split_path + [secret_base]):
+                idna_domain = idna_encode(domain)
```
Comment (lufte) [inline misc/userscripts/qute-pass:162]: This step should go outside the `for path, directories, file_names in os.walk(arguments.password_store, followlinks=True):` loop at line 152, as the domain doesn't change between entries in my password store.

File: misc/userscripts/qute-pass [hunk 2]
```diff
@@ -140,9 +159,13 @@ def find_pass_candidates(domain, unfiltered=False):
...
+                idna_split_path = [idna_encode(part) for part in split_path]
```
Comment (lufte) [inline misc/userscripts/qute-pass:163]: And this one should go outside the `for secret in secrets` loop as the split_path doesn't change between different secrets under the same folder.

**Review by lufte — CHANGES_REQUESTED** [review]
So let me see if I get this right: the idea is to encode the domain of the requested URL, and encode all the domains stored in the pass store, to ensure we're comparing apples to apples. I think this makes sense.

I left some comments on stuff that's being computed over and over with no need.

**Comment by DrTobe** [comment]
> So let me see if I get this right: the idea is to encode the domain of the requested URL, and encode all the domains stored in the pass store, to ensure we're comparing apples to apples. I think this makes sense.

You got that right.

> I left some comments on stuff that's being computed over and over with no need.

I have moved these out of the loops to optimize that.

**Review by lufte — APPROVED** [review]
Looks great now :ok_hand:

**Comment by toofar** [comment]
Thanks @DrTobe for the PR and @lufte for the reivew!

**Pull request merged**